src/openapi_server/apis/reading_api.py

```diff
@@ -64,3 +83,64 @@ async def judgment_uri_metadata_get(
     """Unless the client has `read_unpublished_documents` permission,
     then only metadata for published documents are accessible."""
     ...
+
+
+@router.get(
+    "/list/unpublished",
+    responses={
+        200: {"description": "A list of URLs of unpublished documents in JSON format"},
+    },
+    tags=["Reading"],
+    summary="Get a list of unpublished URLs",
+    response_model_by_alias=True,
+)
+async def list_unpublished_get_get(
+    request: Request,
+    response: Response,
+    token_basic: TokenModel = Security(get_token_basic),
+    page: int = 1,  # should not be 0
+) -> Any:
+    """Unless the client has `read_unpublished_documents` permission,
+    then only metadata for published documents are accessible."""
+    client = client_for_basic_auth(token_basic)
+    if not client.user_can_view_unpublished_judgments():
+        response.status_code = 403
+        return {"status": "Not allowed to see unpublished documents"}
+
+    response = client.advanced_search(
+        page=page,
+        show_unpublished=True,
+        only_unpublished=True,
+    )
+
+    xml = decode_multipart_response(response)
+
+    content_type = request.headers.get("Content-Type") or ""
+    if "application/xml" in content_type:
+        return Response(status_code=200, content=xml, media_type="application/xml")
+
+    root = lxml.etree.fromstring(xml)
+    namespaces = {
+        "search": "http://marklogic.com/appservices/search",
+        "uk": "https://caselaw.nationalarchives.gov.uk/akn",
+        "akn": "http://docs.oasis-open.org/legaldocml/ns/akn/3.0",
+    }
+
+    results = []
+
+    for result in root.xpath("//search:result", namespaces=namespaces):
+        data = {}
+        data["raw_uri"] = unpack_list(result.xpath("./@uri"))
+        data["uri"] = f"/detail?judgment_uri={data['raw_uri'].partition('.xml')[0]}"
```

Comment:
This is the URI format for the editor UI. Will we need to give them the URI for the public UI as well?

Reply:
We've decided that we'll just provide a shortened URI that doesn't have .xml at the end.